joy_teleop/test/test_action_fibonacci.py

```diff
@@ -0,0 +1,136 @@
+#    notice, this list of conditions and the following disclaimer.
+#  * Redistributions in binary form must reproduce the above
+#    copyright notice, this list of conditions and the following
+#    disclaimer in the documentation and/or other materials provided
+#    with the distribution.
+#  * Neither the name of PAL Robotics SL. nor the names of its
+#    contributors may be used to endorse or promote products derived
+#    from this software without specific prior written permission.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+# "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+# LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+# FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+# COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+# INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+# BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+# LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+# CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+# LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+# ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+# POSSIBILITY OF SUCH DAMAGE.
+
+import unittest
+
+import action_tutorials_interfaces.action
+import launch
+import launch_ros
+import launch_testing
+import pytest
+import rclpy
+import sensor_msgs.msg
+
+
+@pytest.mark.rostest
+def generate_test_description():
+    parameters = {}
+    parameters['fibonacci.type'] = 'action'
+    parameters['fibonacci.interface_type'] = 'action_tutorials_interfaces/action/Fibonacci'
+    parameters['fibonacci.action_name'] = '/fibonacci'
+    parameters['fibonacci.buttons'] = [2]
+    parameters['fibonacci.action_goal'] = {'order': 5}
+    joy_teleop_node = launch_ros.actions.Node(
+        package='joy_teleop',
+        node_executable='joy_teleop',
+        output='both',
+        parameters=[parameters])
+
+    return (
+        launch.LaunchDescription([
+            joy_teleop_node,
+            launch_testing.actions.ReadyToTest(),
+        ]),
+        {
+            'joy_teleop_node': joy_teleop_node,
+        }
+    )
+
+
+class TestJoyTeleopActionFibonacci(unittest.TestCase):
+
+    def setUp(self):
+        self.context = rclpy.context.Context()
+        rclpy.init(context=self.context)
+        self.node = rclpy.create_node('test_action_fibonacci', context=self.context)
+        self.executor = rclpy.executors.SingleThreadedExecutor(context=self.context)
+        self.executor.add_node(self.node)
+
+    def tearDown(self):
+        self.node.destroy_node()
+        rclpy.shutdown(context=self.context)
+
+    def test_simple_message(self, launch_service, joy_teleop_node, proc_output):
+        joy_publisher = self.node.create_publisher(
+            msg_type=sensor_msgs.msg.Joy,
+            topic='joy',
+            qos_profile=1,
```

Comment:
@clalancette nit: here and elsewhere, can we use fully qualified QoS from `rclpy.qos`?